### PR TITLE
Trap bad request response when web missing

### PR DESCRIPTION
**Pull request by ransombriggs**
@djcp could you review?

### CODE REVIEW

File: test/commands/apps/errors.js [hunk 1]
```diff
@@ -54,6 +54,41 @@ describe('apps:errors', () => {
...
+
+    return cmd.run({app: 'myapp', flags: {json: false}})
+      .then(() => expect(cli.stdout, 'to be', `No errors on myapp in the last 24 hours
+`))
+      .then(() => expect(cli.stderr, 'to be empty'))
+      .then(() => metrics.done())
+      .then(() => heroku.done())
+  })
+
+  it('propegates other bad request', () => {
```
Comment (djcp) [inline test/commands/apps/errors.js]: *propagates

**Comment by djcp** [comment]
So when a user requests a list of errors, if they don't have a `web` process type we respond with a 400 and "invalid process type" error? 

It feels like the correct fix would be at the API level to emit a better error message, or am I confused about what's going on?  Barring that, this fix seems fine until we fix API to emit a better error.

File: commands/apps/errors.js [hunk 2]
```diff
@@ -40,6 +40,13 @@ function * run (context, heroku) {
       host: 'api.metrics.herokai.com',
       path: `/apps/${context.app}/router-metrics/errors?${DATE}&process_type=web`,
       headers: {Range: ''}
+    }).catch((err) => {
+      if (err.statusCode === 400 && err.body && err.body.message &&
+          err.body.message.startsWith('invalid process_type provided')
```
Comment (djcp) [inline commands/apps/errors.js]: Is this case insensitive? It probably should be, just in case. (ouch, that pun hurt)

Comment (ransombriggs) [inline commands/apps/errors.js]: @djcp make case insensitive in https://github.com/heroku/heroku-apps/pull/136/commits/618d40b6193ac2d005af73ba36a1202f559440b6

File: test/commands/apps/errors.js (hunk 1 above)
Comment (ransombriggs) [inline test/commands/apps/errors.js]: @djcp fixed typo in https://github.com/heroku/heroku-apps/pull/136/commits/618d40b6193ac2d005af73ba36a1202f559440b6

**Comment by ransombriggs** [comment]
@djcp turns out I was wrong about where the error was coming from, the API says that there is a web process type, even if there is not one in the `Procfile`, so short of getting a better attribute to match on from the metrics API this is probably as good as we will get without breaking other things.

**Comment by djcp** [comment]
LGTM

**Pull request merged**